### DISCUSSION TITLE
How to get the shortened names of Weekdays and months

### DISCUSSION THREAD

**Discussion by HolgerGottChristensen** [unicode-org/icu4x#4631] [Q&A] [answered]
I am trying to make a calendar widget, and I need to get the names of the weekdays and months for the current locale. Currently my names are hard-coded.

<img width="726" alt="Screenshot 2024-02-25 at 13 16 08" src="https://github.com/unicode-org/icu4x/assets/11473146/52577590-f9b7-4027-9d14-d12832da332f">

Is it possible to get 2 letter abbreviations for the 7 weekdays for some locale?
Is it possible to get the month names in different long and shortened forms using ICU4x?

**Comment by sffc** [comment, marked as answer]
You can access this information using one of two experimental APIs:

1. https://unicode-org.github.io/icu4x/rustdoc/icu_datetime/options/components/index.html -- request component `weekday` with `Text::Short`
2. https://unicode-org.github.io/icu4x/rustdoc/icu_datetime/neo_pattern/struct.DateTimePattern.html -- create a pattern for a standalone weekday or month name field, which you can find in [the DateTime field table](https://unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table), such as `LLLL` and `ccc`

Note that option 1 is released on crates.io but option 2 is newer code. When we have semantic skeleta (#1317) that will be another option to access this data.

**Reply by HolgerGottChristensen** [reply]
Thank you very much. Just to be sure, I need to create a set of options for a formatter that specifies i only want the weekdays in the format and then I need to give it 7 consecutive days and then I can use the results?
